extension/src/saturation.rs

```diff
@@ -0,0 +1,114 @@
+use pgx::*;
+
+/// Saturating Math for Integers
+
+/// Computes x+y, saturating at the numeric bounds instead of overflowing
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn addition_saturate_to_min(x: i32, y: i32) -> i32 {
+    x.saturating_add(y)
+}
+
+/// Computes x+y, saturating at 0 for the minimum bound instead of i32::MIN
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn addition_saturate_to_zero(x: i32, y: i32) -> i32 {
+    // check to see if abs of y is greater than the abs of x?
+    let result = x.saturating_add(y);
+    if result > 0 {
+        result
+    } else {
+        0
+    }
+}
+
+/// Computes x-y, saturating at the numeric bounds instead of overflowing.
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn subtraction_saturate_to_min(x: i32, y: i32) -> i32 {
+    x.saturating_sub(y)
+}
+
+/// Computes x-y, saturating at 0 for the minimum bound instead of i32::MIN
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn subtraction_saturate_to_zero(x: i32, y: i32) -> i32 {
+    if y > x {
+        0
+    } else {
+        x.saturating_sub(y)
+    }
+}
+
+/// Computes x*y, saturating at the numeric bounds instead of overflowing
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn multiplication_saturate_to_min(x: i32, y: i32) -> i32 {
+    x.saturating_mul(y)
+}
+
+/// Computes x*y, saturating at 0 for the minimum bound instead of i32::MIN
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn multiplication_saturate_to_zero(x: i32, y: i32) -> i32 {
```

Comment:
I don't think we actually need a saturate_to_zero for multiplication.  If a user needs this to be positive they probably need the inputs to this to be positive.  If they've checked that, then the result will necessarily be positive.